### ISSUE TITLE
slurs accelerate Deepspeed launching rises a lot of "WARNING: Skipping ... as it contains forbidden characters or missing values." and the training does not utilize multi node

### ISSUE THREAD

**Issue by ermu2001** [huggingface/accelerate#2839]
### System Info

```Shell
The world dict info is then: 

...
[2024-06-07 20:06:54,249] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [0, 1, 2, 3, 4, 5, 6, 7]}
...

```

not sure whether it is related to the warnings.
```


### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

starting multi node training with script
```
set -x -e
######################
######################
source scripts/exec_env.sh
export head_node_ip=$(scontrol show hostnames $SLURM_JOB_NODELIST | head -n 1)
######################

# SET UP PYTHONPATH FOR IMPORT
echo "PYTHONPATH: ${PYTHONPATH}"
which_python=$(which python)
echo "which python: ${which_python}"
export PYTHONPATH=${PYTHONPATH}:${which_python}
export PYTHONPATH=${PYTHONPATH}:.
echo "PYTHONPATH: ${PYTHONPATH}"

# SHARED ARGUMENTS
export n_node=$SLURM_JOB_NUM_NODES
export num_save_samples=80000
export num_gpus=$[$n_node * 8]
export full_batch_size=128 # total batch size for gradient update
export batch_size=4 # this is the batch size for each single gpu
export save_steps=$[$num_save_samples/($batch_size*$num_gpus)] # save step is base on the "seen data" by the model. As the loop was counted base on data iteration, save step here doesn't take gradient accumulation into consideration.
export ckpt_steps=$[$save_steps/10]
export gradient_accumulation_steps=$[$full_batch_size/($batch_size*$num_gpus)] # some how the gradient accumulation steps should be computed as it was to pass in to Accelerate for not knowing the total GPUs, thus not able to compute afterwards. Align this with your accelerate configuration

######### TRAINING ARGS #############
# Train 7B with LORA
export accel_config=scripts/accel_config_multigpu.yaml # able to run with 80G A100
# export accel_config=scripts/accel_config_deepspeed_zero3_trainLLM7B.yaml # able to run with 80G A100

export LAUNCHER="accelerate launch \
    --num_processes $((SLURM_JOB_NUM_NODES * GPUS_PER_NODE)) \
    --num_machines $SLURM_JOB_NUM_NODES \
    --machine_rank $SLURM_PROCID \
    --rdzv_backend c10d \
    --main_process_ip $head_node_ip \
    --main_process_port 6877 \
    --config_file ${accel_config} \
    "
export SCRIPT="train.py"
export SCRIPT_ARGS=" \
    save_steps $save_steps \
    ckpt_steps $ckpt_steps \
    gradient_accumulation_steps $gradient_accumulation_steps \
    batch_size $batch_size 
"

export CMD="$LAUNCHER $SCRIPT $SCRIPT_ARGS" 

echo "*************************PRE EXECUTION*******************************"
echo full batch size: $full_batch_size
echo instance batch size: $batch_size
echo num gpus: $num_gpus
echo gradient accumulation steps: $gradient_accumulation_steps
echo config file: $config_file
echo accelerate config file: $accel_config
echo "*********************************************************************"

# This step is necessary because accelerate launch does not handle multiline arguments properly
$CMD

```

### Expected behavior

training with multi-gpus does not have such problems, the multi-node training was fine in that situation.

Can anyone indicate what might have gone wrong here. Please let me know if any more information is needed.

**Comment by ermu2001** [comment]
I think I have figured out how to solve the problem here.

It seems that launching with deepspeed in my situation requires using the standard distributed launcher of deepspeed
```yaml
compute_environment: LOCAL_MACHINE
distributed_type: MULTI_GPU
debug: true
deepspeed_config:
 deepspeed_config_file: scripts/zero3_config_34B.json
 deepspeed_multinode_launcher: standard
 zero3_init_flag: true
distributed_type: DEEPSPEED
fsdp_config: {}
machine_rank: 0
main_process_ip: null
main_process_port: null
main_training_function: main
gpu_ids: 0,1,2,3,4,5,6,7
num_machines: 1
num_processes: 8
use_cpu: false
```

And the exporting skipping warning then disappeared and the multimode training works as expected. Though not fully understanding what have gone on switching to "standard" launcher. I'll close this issue soon.

**Comment by dongzhuoyao** [comment]
Got this issue as well, super weird
